### PR TITLE
Hard Delete Participant Data

### PR DESCRIPTION
**Pull request by giuliacampana**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "immutable": "4.0.0-rc.9",
     "lattice": "0.56.2",
```
Comment (UnsungHero97) [inline package.json]: You'll need `0.57.0` to get the new api.

File: src/containers/study/StudySagas.js [hunk 2]
```diff
@@ -530,6 +530,85 @@ function* changeEnrollmentStatusWorker(action :SequenceAction) :Generator<*, *,
...
+     */
+
+    const filter = {
+      entityKeyIds: [participantEntityKeyId],
+      sourceEntitySetIds: [devicesEntitySetId, applicationDataEntitySetId, preprocessedDataEntitySetId],
+    };
+
+    response = yield call(
+      deleteEntitiesAndNeighborsWorker,
+      deleteEntitiesAndNeighbors({ entitySetId: participantsEntitySetId, filter, deleteType: 'Hard' })
```
Comment (UnsungHero97) [inline src/containers/study/StudySagas.js]: This constant is available in `lattice`:

```
import { Types } from 'lattice';
const { DeleteTypes } = Types;
...
deleteEntitiesAndNeighbors({ entitySetId: participantsEntitySetId, filter, deleteType: DeleteTypes.Hard })
```

File: src/containers/study/StudySagas.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  describe('hardDeleteParticipantDataWatcher', () => {
+    testShouldBeGeneratorFunction(hardDeleteParticipantDataWatcher);
+    testWatcherSagaShouldTakeEvery(
+      hardDeleteParticipantDataWatcher,
+      hardDeleteParticipantDataWorker,
+      HARD_DELETE_PARTICIPANT_DATA,
+    );
+  });
+
+  describe('hardDeleteParticipantDataWorker', () => {
```
Comment (UnsungHero97) [inline src/containers/study/StudySagas.test.js]: This is a great first step. As a follow up, you should add a test to cover the `filter` param:
```
const filter = {
  entityKeyIds: [participantEntityKeyId],
  sourceEntitySetIds: [devicesEntitySetId, applicationDataEntitySetId, preprocessedDataEntitySetId],
};
```
We want to be sure that exactly those parameters are being passed along to the API.

File: package.json (hunk 1 above)
Comment (giuliacampana) [reply package.json]: Oops, this must have been deleted somewhere. Thanks!

File: src/containers/study/StudySagas.js (hunk 2 above)
Comment (giuliacampana) [reply src/containers/study/StudySagas.js]: Thanks!

File: src/containers/study/StudySagas.test.js (hunk 3 above)
Comment (giuliacampana) [reply src/containers/study/StudySagas.test.js]: That makes sense! Would you recommend expanding the `testWorkerSagaShouldHandleSuccessCase` function to run through the entire saga in order to test the filter?

**Pull request merged**